spark/src/main/scala/org/apache/spark/ml/regression/SmileRegressor.scala

```diff
@@ -0,0 +1,241 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2020 Haifeng Li. All rights reserved.
+ *
+ * Smile is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * Smile is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Smile.  If not, see <https://www.gnu.org/licenses/>.
+ ******************************************************************************/
+
+package org.apache.spark.ml.regression
+
+import java.io.{ObjectInputStream, ObjectOutputStream}
+
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.SparkContext
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.util.Instrumentation._
+import org.apache.spark.ml.util.{Identifiable, _}
+import org.apache.spark.ml.{Predictor, PredictorParams}
+import org.apache.spark.sql.Dataset
+import org.apache.spark.storage.StorageLevel
+import org.json4s.JObject
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods._
+
+private[ml] trait SmileRegressorParams extends Params with PredictorParams {
```

Comment:
I don't see difference between SmileClassificationParams and SmileRegressorParams. Do we need both?

Reply:
It's different because the trainer params is either ... => smile.regression.Regression[Array[Double]] or ... =>  smile.classification.Classifier[Array[Double]]